tests/syntax/test_pruning.py

```diff
@@ -208,39 +207,49 @@ def test_visibility_pruning():
     assert all(pos.distanceTo(Vector(0, 0, 0)) <= 1.1 for pos in positions)
     assert any(pos.distanceTo(Vector(0, 0, 0)) >= 1 for pos in positions)
 
-
-def test_visibility_pruning_cyclical():
-    """A case where a cyclical dependency could be introduced if pruning is not done carefully.
-
-    NOTE: We don't currently prune this case so this test is a sentinel for future behavior.
-    """
+    # requireVisible with offset
+    baseOffsetVal = 0.0001
     scenario = compileScenic(
-        """
-        workspace = Workspace(PolygonalRegion([0@0, 100@0, 100@100, 0@100]))
-        foo = new Object with requireVisible True, in workspace
-        ego = new Object visible from foo, in workspace
+        f"""
+        workspace = Workspace(RectangularRegion(0@0, 0, 1e10, 1e10))
+        ego = new Object at (0,0,0), with visibleDistance 1
+        foo = new Object on workspace, with requireVisible True,
+            with shape SpheroidShape(dimensions=(0.2,0.2,0.2)),
+            with baseOffset (0,0,{baseOffsetVal}), with contactTolerance 0
+        param p = foo.position
         """
     )
+    positions = [sampleParamP(scenario, maxIterations=100) for i in range(30)]
+    assert all(pos.distanceTo(Vector(0, 0, 0)) <= 1.1 for pos in positions)
+    assert any(pos.distanceTo(Vector(0, 0, 0)) >= 1 for pos in positions)
```

Comment:
My math may be off, but I'm getting a probability of about 1.6e-05 for this assertion to fail, which is too large (I usually shoot for 1e-18).

Reply:
Fixed as discussed over Slack.